### PR TITLE
add sanitizer

### PR DESCRIPTION
**Pull request by skyzh**
Signed-off-by: Alex Chi <iskyzh@gmail.com>

fix https://github.com/tikv/agatedb/issues/10

Sanitizer will only run when there's any change in `skiplist` module. First successful run is triggered along with `split sanitizer` commit.

In the future, we could apply sanitizer to other modules.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -13,13 +10,16 @@ clippy:
...
+	RUSTFLAGS="$(SANITIZER_FLAGS)" cargo test --all-features --workspace
+
 bench:
 	cargo bench --all-features --workspace
 
-ci: fmt clippy test run_release
+bench_sanitizer:
+	RUSTFLAGS="$(SANITIZER_FLAGS)" cargo bench --all-features --workspace
 
-proto: proto/proto/meta.proto proto/proto/rustproto.proto
```
Comment (BusyJay) [inline Makefile:21]: Why remove this?

**Review by BusyJay — COMMENTED** [review]
Why limit to skiplist only?

File: Makefile (hunk 1 above)
Comment (skyzh) [reply Makefile:21]: Now we use prost-build to generate `meta.rs` on cargo build, instead of generating them manually.

**Comment by skyzh** [comment]
> Why limit to skiplist only?

Run all sanitizer bench takes ~40mins, and unsafe code is mostly used in skiplists. So I choose to run sanitizer bench only on there's any change in skiplist.

Maybe we could run this bench manually instead of triggering by PR or commit, and I could include all modules in bench?

**Comment by BusyJay** [comment]
> Maybe we could run this bench manually

Seems more practical.

**Comment by skyzh** [comment]
I've updated the CI config. After this PR being merged, we could manually trigger sanitizer test and bench on GitHub Action webpage.

**Review by BusyJay — APPROVED** [review]

**Comment by skyzh** [comment]
image

Now benchmark can be triggered in Action tab

**Pull request merged**